packages/anon-aadhaar-react/src/util.ts

```diff
@@ -62,40 +12,91 @@ export function text(emoji: string, text: string) {
  * @param pdf pdf buffer
  * @returns {Signature, signedData}
  */
-export const uploadPdf = (
+export const uploadQRpng = (
   e: ChangeEvent<HTMLInputElement>,
-  setpdfStatus: Dispatch<SetStateAction<null | AadhaarPdfValidation>>,
-): Promise<{ pdf: Buffer }> => {
+  setQrStatus: Dispatch<SetStateAction<AadhaarQRValidation | null>>,
+): Promise<{ qrValue: string }> => {
   return new Promise((resolve, reject) => {
     if (e.target.files) {
       try {
         const fileReader = new FileReader()
-        fileReader.readAsBinaryString(e.target.files[0])
+        fileReader.readAsDataURL(e.target.files[0])
         fileReader.onload = e => {
-          if (e.target) {
+          if (e.target && e.target.result) {
             try {
-              const { signature } = extractSignature(
-                Buffer.from(e.target.result as string, 'binary'),
-              )
+              const image = new Image()
+              image.onload = () => {
+                const canvas = document.createElement('canvas')
+                canvas.width = image.width
+                canvas.height = image.height
+                const ctx = canvas.getContext('2d')
+
+                if (!ctx) throw Error('Image cannot be reconstructed')
+
+                ctx.drawImage(image, 0, 0)
+                const imageData = ctx.getImageData(
+                  0,
+                  0,
+                  image.width,
+                  image.height,
+                )
+
+                const qrValue = jsQR(imageData.data, image.width, image.height)
 
-              if (signature != '') {
-                resolve({
-                  pdf: Buffer.from(e.target.result as string, 'binary'),
-                })
-                setpdfStatus(AadhaarPdfValidation.SIGNATURE_PRESENT)
-              } else {
-                setpdfStatus(AadhaarPdfValidation.SIGNATURE_NOT_PRESENT)
+                if (qrValue != null) {
+                  resolve({
+                    qrValue: qrValue.data,
+                  })
+                  setQrStatus(AadhaarQRValidation.QR_CODE_SCANNED)
+                } else {
+                  setQrStatus(AadhaarQRValidation.ERROR_PARSING_QR)
+                }
               }
+              image.src = e.target.result.toString()
             } catch (error) {
-              setpdfStatus(AadhaarPdfValidation.ERROR_PARSING_PDF)
+              setQrStatus(AadhaarQRValidation.ERROR_PARSING_QR)
+              console.error(error)
               reject(error)
             }
           }
         }
       } catch {
-        setpdfStatus(null)
-        reject()
+        setQrStatus(null)
+        reject(new Error('No file selected'))
       }
     }
   })
 }
+
+/**
+ * Fetch the public key certificate from the serverless function endpoint.
+ * @param url Endpoint URL to fetch the public key.
+ * @returns {Promise<string | null>} The official Aadhaar public key.
```

Comment:
Do this return aadhaar certificate?

Reply:
Yes it fetches either test certificate or the official certificate file, I changed the name for better understanding.